### PR TITLE
fix: Return only accepted session and public speaker call by default

### LINKED ISSUE #5998: Fetching featured speakers, accepted sessions, public call for speaker by default
**Describe the bug**
By default, the server should only return featured speakers, accepted sessions, public speakers call when client-side make GET requests. Currently, it also returns pending sessions or private call for speaker.

**Additional context**
Discussed in: https://github.com/fossasia/open-event-android/pull/1902 (comment)

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5998 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/sessions.py [hunk 1]
```diff
@@ -89,22 +89,24 @@ def query(self, view_kwargs):
         query_ = self.session.query(Session)
```
Comment (iamareebjamal) [inline app/api/sessions.py]: Why write it 4 times when it can be written once

File: app/api/speakers_calls.py [hunk 2]
```diff
@@ -86,20 +86,29 @@ def before_get_object(self, view_kwargs):
         """
         if view_kwargs.get('event_identifier'):
             try:
-                event = self.session.query(Event).filter_by(identifier=view_kwargs['event_identifier']).one()
+                event = self.session.query(Event).filter_by(identifier=view_kwargs['event_identifier'],
+                                                            privacy='public').one()
```
Comment (iamareebjamal) [inline app/api/speakers_calls.py]: public and accepted

Comment (prateekj117) [reply app/api/speakers_calls.py]: Speaker calls are an invitation to submit a proposal for being a speaker at the event. They are either `private` or `public`. I didn't get the context of accepted here. Did you mean making changes for speakers to only return accepted speakers?

File: app/api/sessions.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ def query(self, view_kwargs):
         :param view_kwargs:
         :return:
         """
-        query_ = self.session.query(Session)
+        query_ = self.session.query(Session).filter(Session.state == "accepted")
```
Comment (iamareebjamal) [inline app/api/sessions.py]: deleted_at=None

Comment (prateekj117) [reply app/api/sessions.py]: @iamareebjamal I tested it, in this particular case as well, it doesn't return rows with `deleted_at != null`

Comment (iamareebjamal) [reply app/api/sessions.py]: How does it do that automatically?

Comment (prateekj117) [reply app/api/sessions.py]: @iamareebjamal I will debug which part of the code does that, but I tested it, it was not returning deleted sessions.

Comment (prateekj117) [reply app/api/sessions.py]: @iamareebjamal This is filtered by flask-rest-jsonapi by default. 

https://github.com/fossasia/flask-rest-jsonapi/blob/4b10b48e443a839bbff369c769198ede34fdbff3/flask_rest_jsonapi/data_layers/alchemy.py#L94

Comment (iamareebjamal) [reply app/api/sessions.py]: OK :+1:

**Pull request merged**